main/inc/lib/tracking.lib.php

```diff
@@ -3347,6 +3347,104 @@ public static function count_student_exercise_attempts(
         return $count_attempts;
     }
 
+    /**
+     * It gets the last finalization date of learnpaths in a course.
+     *
+     * @param int  $userId
+     * @param int  $courseId
+     * @param int  $sessionId
+     * @param bool $convertDate
+     *
+     * @return false|mixed|string
+     */
+    public static function getCourseLpFinalizationDate(
+        $userId,
+        $courseId,
+        $sessionId,
+        $convertDate = true
+    ) {
+        $tblLpView = Database::get_course_table(TABLE_LP_VIEW);
+        $tblLpItem = Database::get_course_table(TABLE_LP_ITEM);
+        $tblLpItemView = Database::get_course_table(TABLE_LP_ITEM_VIEW);
+
+        $courseId = (int) $courseId;
+        $userId = (int) $userId;
+        $sessionId = (int) $sessionId;
```

Comment:
If you use typed parameters, then these 3 lines can go away because the parameters will be forced to this type (int).